.circleci/config.yml

```diff
@@ -0,0 +1,216 @@
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-npm-deps-{{ checksum "package.json" }}
+
+      - run:
+          name: Run tests
+          command: |
+            source venv/bin/activate
+            pip install -r requirements/edx/local.txt
+            paver test_system -s lms --with-flaky --cov-args="-p" --with-xunitmp
+            echo "---------- DONE ----------"
+
+  cms_unit_tests:
+    docker:
+      - image: edxops/xenial-common:hawthorn.beta1
+    environment:
+      - NO_PREREQ_INSTALL: "true"
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+
+      - run:
+          name: Do base configuration
+          command: |
+            ./scripts/circle-ci-configuration.sh
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-{{ checksum "requirements/edx/pre.txt" }}-{{ checksum "requirements/edx/paver.txt" }}-{{ checksum "requirements/edx/github.txt" }}-{{ checksum "requirements/edx/base.txt" }}-{{ checksum "requirements/edx/local.txt" }}-{{ checksum "requirements/edx/post.txt" }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-npm-deps-{{ checksum "package.json" }}
+
+      - run:
+          name: Run tests
+          command: |
+            source venv/bin/activate
+            pip install -r requirements/edx/local.txt
+            paver test_system -s cms --with-flaky --cov-args="-p" --with-xunitmp
+            echo "---------- DONE ----------"
+
+  lib_unit_tests:
+    docker:
+      - image: edxops/xenial-common:hawthorn.beta1
+    environment:
+      - NO_PREREQ_INSTALL: "true"
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+
+      - run:
+          name: Do base configuration
+          command: |
+            ./scripts/circle-ci-configuration.sh
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-{{ checksum "requirements/edx/pre.txt" }}-{{ checksum "requirements/edx/paver.txt" }}-{{ checksum "requirements/edx/github.txt" }}-{{ checksum "requirements/edx/base.txt" }}-{{ checksum "requirements/edx/local.txt" }}-{{ checksum "requirements/edx/post.txt" }}
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-npm-deps-{{ checksum "package.json" }}
+
+      - run:
+          name: Run tests
+          command: |
+            source venv/bin/activate
+            pip install -r requirements/edx/local.txt
+            paver test_lib --with-flaky --cov-args="-p" --with-xunitmp
+            echo "---------- DONE ----------"
```

Comment:
nit: I don't think we need this echo here, CircleCI UI is pretty clear in regards to the commands that are run.